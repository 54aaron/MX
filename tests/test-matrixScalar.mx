int main(){
    Matrix d;
    Matrix n;
    Matrix q;
    Matrix s;
    Matrix o;
    Matrix p;
    d = [[1,2,3],[4,5,6], [7,8,9]];
    n = d**.2;
    q = [[1,2,3],[4,5,6]];
    s = q **.0;
    o = identity(4);
    p = o **.-9;
    prints("==============================================");
    prints("Matrix d = ");
    print_matrix(d);
    prints("==============================================");
    prints("Matrix d**.2 = ");
    print_matrix(n);
    prints("==============================================");
    prints("Matrix q = ");
    print_matrix(q);
    prints("==============================================");
    prints("Matrix q**.0 = ");
    print_matrix(s);
    prints("==============================================");
    prints("Matrix o = ");
    print_matrix(o);
    prints("==============================================");
    prints("Matrix o**.-9 = ");
    print_matrix(p);
    prints("==============================================");
}
