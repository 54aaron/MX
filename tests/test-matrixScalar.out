==============================================
Matrix d = 

[ 1, 2, 3 ]
[ 4, 5, 6 ]
[ 7, 8, 9 ]
==============================================
Matrix d**.2 = 

[ 2, 4, 6 ]
[ 8, 10, 12 ]
[ 14, 16, 18 ]
==============================================
Matrix q = 

[ 1, 2, 3 ]
[ 4, 5, 6 ]
==============================================
Matrix q**.0 = 

[ 0, 0, 0 ]
[ 0, 0, 0 ]
==============================================
Matrix o = 

[ 1, 0, 0, 0 ]
[ 0, 1, 0, 0 ]
[ 0, 0, 1, 0 ]
[ 0, 0, 0, 1 ]
==============================================
Matrix o**.-9 = 

[ -9, 0, 0, 0 ]
[ 0, -9, 0, 0 ]
[ 0, 0, -9, 0 ]
[ 0, 0, 0, -9 ]
==============================================
